Sources/ArcGISToolkit/Components/Authentication/Authenticator.swift

```diff
@@ -50,7 +50,11 @@ extension Authenticator: ArcGISAuthenticationChallengeHandler {
         
         // Create the correct challenge type.
         if let configuration = oAuthUserConfigurations.first(where: { $0.canBeUsed(for: challenge.requestURL) }) {
-            return .continueWithCredential(try await OAuthUserCredential.credential(for: configuration))
+            do {
+                return .continueWithCredential(try await OAuthUserCredential.credential(for: configuration))
+            } catch is CancellationError {
+                return .cancel
+            }
```

Comment:
I don't understand why this change is needed

Reply:
So the request will fail with `ArcGISChallengeCancellationError with underlying error` instead of `CancellationError`.

Reply:
should It be catching `ArcGISChallengeCancellationError` then?

Reply:
Never mind, I think I understand. I do think a comment would be good here probably

Reply:
done